addons/pos_restaurant/static/src/js/models.js

```diff
@@ -236,18 +235,8 @@ patch(PosGlobalState.prototype, "pos_restaurant.PosGlobalState", {
         return tableOrders.reduce((count, order) => count + order.getCustomerCount(), 0);
     },
     create_printer(config) {
-        var url = config.proxy_ip || "";
-        if (url.indexOf("//") < 0) {
-            url = window.location.protocol + "//" + url;
-        }
-        if (url.indexOf(":", url.indexOf("//") + 2) < 0 && window.location.protocol !== "https:") {
-            url = url + ":8069";
-        }
-        return new HWPrinter({
-            connection: new Session(undefined, url || "http://localhost:8069", {
-                use_cors: true,
-            }),
-        });
+        const url = deduceUrl(config.proxy_ip || "");
+        return new HWPrinter({ rpc: this.hardwareProxy.rpc, url });
```

Comment:
I can't directly pass the hardwareProxy because the `config.proxy_ip` can be different from the `ip` of the hardwareProxy.